### PR TITLE
Add doc-strings for unsigned indices model classes

### PR DESCRIPTION
**Pull request by rohanshah18**
## Problem

The java sdk contains custom model classes created to support the conversion of unsigned 32 bit integers to signed 32 bit integers and vice-versa, which are currently missing the doc-strings.

## Solution

Add doc strings to the following model classes:
1. QueryResponseWithUnsignedIndices
2. ScoredVectorWithUnsignedIndices
3. SparseValuesWithUnsignedIndices
4. VectorWithUnsignedIndices

## Type of Change

- [X] Non-code change (docs, etc)

## Test Plan

CI tests

### CODE REVIEW

File: src/main/java/io/pinecone/unsigned_indices_model/QueryResponseWithUnsignedIndices.java [hunk 1]
```diff
@@ -8,13 +8,43 @@
 import java.util.Collections;
 import java.util.List;
 
+/**
+ * This class represents the response to a query, where the `ScoredVector` objects in the `matches` list
+ * contain `SparseValuesWithUnsignedIndices` instead of the standard `SparseValues`.
+ * <p>
+ * The `QueryResponseWithUnsignedIndices` class contains the following fields:
+ * - `matches`: a list of `ScoredVectorWithUnsignedIndices` objects, representing the matching vectors
+ * - `namespace`: the namespace of the query
```
Comment (aulorbe) [inline src/main/java/io/pinecone/unsigned_indices_model/QueryResponseWithUnsignedIndices.java:17]: `"the namespace of the query"` is a little confusingly worded to me. Maybe you could write something more like `"the namespace in your index that you want to retrieve search results from"`?

File: src/main/java/io/pinecone/unsigned_indices_model/QueryResponseWithUnsignedIndices.java [hunk 2]
```diff
@@ -8,13 +8,43 @@
...
+ * - `usage`: the usage information for the query
```
Comment (aulorbe) [inline src/main/java/io/pinecone/unsigned_indices_model/QueryResponseWithUnsignedIndices.java:18]: Defining a word by using the word (`usage`) is a little confusing. Looking at this, I'm not sure what `"usage"` means here. Does this mean, like, read and write units?

File: src/main/java/io/pinecone/unsigned_indices_model/QueryResponseWithUnsignedIndices.java [hunk 3]
```diff
@@ -8,13 +8,43 @@
...
+ * <p>
+ * The class provides a constructor that takes a `QueryResponse` object and converts it to a
+ * `QueryResponseWithUnsignedIndices` object, as well as getter and setter methods for each of the fields.
+ * It also includes a method to convert a list of `ScoredVector` objects to a list of
+ * `ScoredVectorWithUnsignedIndices` objects.
+ */
```
Comment (aulorbe) [inline src/main/java/io/pinecone/unsigned_indices_model/QueryResponseWithUnsignedIndices.java:24]: This might be convention, so ignore me if it is, but this seems like too much info for a docstring. I think you can just define the class at a high level and define the params. Plus, outlining each method like this in a docstring means it'll be a pain to update when we add new ones/remove old ones.

File: src/main/java/io/pinecone/unsigned_indices_model/QueryResponseWithUnsignedIndices.java [hunk 4]
```diff
@@ -27,14 +57,32 @@ public QueryResponseWithUnsignedIndices(QueryResponse queryResponse) {
         }
     }
 
+    /**
+     * Returns the list of matching vectors, where each vector contains sparse values with unsigned 32-bit integer indices.
```
Comment (aulorbe) [inline src/main/java/io/pinecone/unsigned_indices_model/QueryResponseWithUnsignedIndices.java:61]: I didn't know what `"unsigned"` meant until I joined this team. Might be worth defining in the class or here in this method that `"unsigned"` means w/o a negative or positive sign (if that's what it means? I'm still not 100% sure honestly....)

File: src/main/java/io/pinecone/unsigned_indices_model/QueryResponseWithUnsignedIndices.java [hunk 5]
```diff
@@ -27,14 +57,32 @@ public QueryResponseWithUnsignedIndices(QueryResponse queryResponse) {
...
+     * @return the `ScoredVectorWithUnsignedIndices` object at the specified index
+     */
     public ScoredVectorWithUnsignedIndices getMatches(int index) {
         return matches.get(index);
     }
 
+    /**
+     * Converts a list of `ScoredVector` objects to a list of `ScoredVectorWithUnsignedIndices` objects.
+     *
+     * @param matches the list of `ScoredVector` objects to convert
```
Comment (aulorbe) [inline src/main/java/io/pinecone/unsigned_indices_model/QueryResponseWithUnsignedIndices.java:82]: Colon btwn `@param matches` and `the`?

File: src/main/java/io/pinecone/unsigned_indices_model/QueryResponseWithUnsignedIndices.java [hunk 6]
```diff
@@ -46,25 +94,56 @@ public List<ScoredVectorWithUnsignedIndices> convertToScoredVectorWithUnsignedIn
...
+     * Sets the list of matching vectors, where each vector contains sparse values with unsigned 32-bit integer indices.
+     *
+     * @param matches the new list of `ScoredVectorWithUnsignedIndices` objects
+     */
     public void setMatches(List<ScoredVectorWithUnsignedIndices> matches) {
         this.matches = matches;
     }
 
+    /**
+     * Returns the namespace of the query.
```
Comment (aulorbe) [inline src/main/java/io/pinecone/unsigned_indices_model/QueryResponseWithUnsignedIndices.java:107]: Again, I'd reword the namespace stuff here. I feel like it sounds like the query belongs to a namespace, which it doesn't; it's just routed to a namespace within your target index

File: src/main/java/io/pinecone/unsigned_indices_model/QueryResponseWithUnsignedIndices.java [hunk 7]
```diff
@@ -46,25 +94,56 @@ public List<ScoredVectorWithUnsignedIndices> convertToScoredVectorWithUnsignedIn
...
+     *
+     * @param namespace the new namespace
+     */
     public void setNamespace(String namespace) {
         this.namespace = namespace;
     }
 
-    public Usage getUsage() { return usage; }
+    /**
+     * Returns the usage information for the query.
```
Comment (aulorbe) [inline src/main/java/io/pinecone/unsigned_indices_model/QueryResponseWithUnsignedIndices.java:125]: Same comment about `usage` above

File: src/main/java/io/pinecone/unsigned_indices_model/ScoredVectorWithUnsignedIndices.java [hunk 8]
```diff
@@ -6,14 +6,55 @@
...
+ * This class represents a scored vector with unsigned 32-bit integer indices for the sparse values.
+ * Unlike the `ScoredVector` class, which uses the `SparseValues` class to represent the sparse values,
+ * this class uses the `SparseValuesWithUnsignedIndices` class, which allows for the use of unsigned 32-bit
+ * integers as indices.
+ * <p>
+ * The `ScoredVectorWithUnsignedIndices` class contains the following fields:
+ * - `score`: the score associated with the vector
+ * - `id`: the identifier of the vector
+ * - `values`: the dense vector values
+ * - `metadata`: the metadata associated with the vector
```
Comment (aulorbe) [inline src/main/java/io/pinecone/unsigned_indices_model/ScoredVectorWithUnsignedIndices.java:19]: Might be nice to give an example of what `metadata` looks like here

File: src/main/java/io/pinecone/unsigned_indices_model/SparseValuesWithUnsignedIndices.java [hunk 9]
```diff
@@ -6,22 +6,55 @@
 import java.util.List;
 
 import static io.pinecone.utils.SparseIndicesConverter.convertSigned32IntToUnsigned32Int;
-
+/**
+ * This class represents a set of sparse values, where the indices are represented as unsigned 32-bit integers.
+ * Unlike the `SparseValues` class, which uses signed 32-bit integers for the indices, this class uses `Long` to
```
Comment (aulorbe) [inline src/main/java/io/pinecone/unsigned_indices_model/SparseValuesWithUnsignedIndices.java:11]: `uses Long objects to represent...` I'd say, instead of just `Long`. Sounds weird since you say "signed `integers`" before.

File: src/main/java/io/pinecone/unsigned_indices_model/SparseValuesWithUnsignedIndices.java [hunk 10]
```diff
@@ -6,22 +6,55 @@
...
+ * represent the indices, allowing for the full range of unsigned 32-bit integers (0 to 4,294,967,295).
+ * <p>
+ * The `indicesWithUnsigned32Int` list contains the indices, while the `values` list contains the corresponding
+ * values. The two lists are parallel, meaning that the value at index `i` in the `values` list corresponds to
```
Comment (aulorbe) [inline src/main/java/io/pinecone/unsigned_indices_model/SparseValuesWithUnsignedIndices.java:15]: This might just be me, so feel free to ignore if so, but I haven't heard of lists being called "parallel" before. Maybe just say the lists are of equal length and order?

**Review by aulorbe — APPROVED** [review]
Nothing blocking, but I think some clarification would be good before merging. See comments :)

File: src/main/java/io/pinecone/unsigned_indices_model/QueryResponseWithUnsignedIndices.java [hunk 11]
```diff
@@ -8,13 +8,43 @@
 import java.util.Collections;
 import java.util.List;
 
+/**
+ * This class represents the response to a query, where the `ScoredVector` objects in the `matches` list
+ * contain `SparseValuesWithUnsignedIndices` instead of the standard `SparseValues`.
+ * <p>
+ * The `QueryResponseWithUnsignedIndices` class contains the following fields:
```
Comment (austin-denoble) [inline src/main/java/io/pinecone/unsigned_indices_model/QueryResponseWithUnsignedIndices.java:15]: I'm not sure if the backticks work for the documentation here. I tried it using the `gradle generateJavadoc` command and it didn't seem to apply any styling.

You can use the `{@link QueryResponseWithUnsignedIndices}` syntax for things that are definitions of classes to provide actual documentation links.

File: src/main/java/io/pinecone/unsigned_indices_model/QueryResponseWithUnsignedIndices.java [hunk 12]
```diff
@@ -8,13 +8,43 @@
...
+    /**
+     * The usage information for the query.
+     */
+    private Usage usage;
 
+    /**
+     * Constructs a `QueryResponseWithUnsignedIndices` object from a `QueryResponse` object, converting the
+     * `ScoredVector` objects to `ScoredVectorWithUnsignedIndices` objects.
+     *
+     * @param queryResponse the `QueryResponse` object to convert
```
Comment (austin-denoble) [inline src/main/java/io/pinecone/unsigned_indices_model/QueryResponseWithUnsignedIndices.java:46]: **nitpick:** I've been trying to capitalize the first word after the param name in these param lists, to differentiate things visually a bit more.

File: src/main/java/io/pinecone/unsigned_indices_model/QueryResponseWithUnsignedIndices.java [hunk 13]
```diff
@@ -27,14 +57,32 @@ public QueryResponseWithUnsignedIndices(QueryResponse queryResponse) {
...
+     * @return the list of `ScoredVectorWithUnsignedIndices` objects
+     */
     public List<ScoredVectorWithUnsignedIndices> getMatchesList() {
         return matches;
     }
 
+    /**
+     * Returns the `ScoredVectorWithUnsignedIndices` object at the specified index in the `matches` list.
+     *
+     * @param index the index of the `ScoredVectorWithUnsignedIndices` object to return
```
[the remark this answers is not in the document]
Comment (austin-denoble) [reply src/main/java/io/pinecone/unsigned_indices_model/QueryResponseWithUnsignedIndices.java:72]: I think we can just capitalize the first word after the param name, at least I was seeing that in examples in the wild.

**Review by austin-denoble — APPROVED** [review]
Looks fine overall, thank you for adding these. A couple nitpick / style suggestions, up to you in terms of iterating.

File: src/main/java/io/pinecone/unsigned_indices_model/QueryResponseWithUnsignedIndices.java (hunk 2 above)
Comment (rohanshah18) [reply src/main/java/io/pinecone/unsigned_indices_model/QueryResponseWithUnsignedIndices.java:18]: Yes

File: src/main/java/io/pinecone/unsigned_indices_model/QueryResponseWithUnsignedIndices.java (hunk 3 above)
Comment (rohanshah18) [reply src/main/java/io/pinecone/unsigned_indices_model/QueryResponseWithUnsignedIndices.java:24]: Yeah but these are custom classes which are used for converting from unsigned 32-bit integers to signed 32-bit integers and vice-versa. There are also a correspond set of classes like ScoredVector, QueryResponse, SparseValues, and Vector generated using protoc command which represents sparse indices in signed 32 bit format.

File: src/main/java/io/pinecone/unsigned_indices_model/QueryResponseWithUnsignedIndices.java (hunk 4 above)
Comment (rohanshah18) [reply src/main/java/io/pinecone/unsigned_indices_model/QueryResponseWithUnsignedIndices.java:61]: I have tried explaining it in SparseValuesWithUnsignedIndices class:
```
 * This class represents a set of sparse values, where the indices are represented as unsigned 32-bit integers.
 * Unlike the `SparseValues` class, which uses signed 32-bit integers for the indices, this class uses `Long` to
 * represent the indices, allowing for the full range of unsigned 32-bit integers (0 to 4,294,967,295).
```
